### PR TITLE
cmd/tailscale/cli: [up] reuse --advertise-tags for OAuth key generation

### PR DESCRIPTION
**Pull request by maisem**
We need to always specify tags when creating an AuthKey from an OAuth key.

Check for that, and reuse the `--advertise-tags` param.

Updates #7982

### CODE REVIEW

**Review by willnorris — APPROVED** [review]

File: cmd/tailscale/cli/up.go [hunk 1]
```diff
@@ -1193,7 +1192,7 @@ func resolveAuthKey(ctx context.Context, v string) (string, error) {
 				Reusable:      false,
 				Ephemeral:     ephemeral,
 				Preauthorized: preauth,
-				Tags:          tags,
+				Tags:          strings.Split(tags, ","),
```
Comment (bradfitz) [inline cmd/tailscale/cli/up.go:1195]: https://go.dev/play/p/mJfrOtwq6VS

File: cmd/tailscale/cli/up.go [hunk 2]
```diff
@@ -1166,10 +1169,6 @@ func resolveAuthKey(ctx context.Context, v string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	var tags []string
-	if v := attrs.Get("tags"); v != "" {
-		tags = strings.Split(v, ",")
-	}
```
Comment (bradfitz) [inline cmd/tailscale/cli/up.go:1172]: that's why I did this

File: cmd/tailscale/cli/up.go [hunk 3]
```diff
@@ -1121,19 +1121,22 @@ func init() {
...
+func resolveAuthKey(ctx context.Context, v, tags string) (string, error) {
 	if !strings.HasPrefix(v, "tskey-client-") {
 		return v, nil
 	}
 	if !envknob.Bool("TS_EXPERIMENT_OAUTH_AUTHKEY") {
 		return "", errors.New("oauth authkeys are in experimental status")
 	}
+	if tags == "" {
+		return "", errors.New("oauth authkeys require --advertise-tags")
+	}
```
Comment (bradfitz) [inline cmd/tailscale/cli/up.go:1139]: ah

**Review by bradfitz — APPROVED** [review]

File: cmd/tailscale/cli/up.go [hunk 4]
```diff
@@ -1121,19 +1121,22 @@ func init() {
...
 //
-//	tskey-client-xxxx[?ephemeral=false&tags=foo,bar&preauthorized=BOOL&baseURL=...]
+//	tskey-client-xxxx[?ephemeral=false&bar&preauthorized=BOOL&baseURL=...]
 //
 // and does the OAuth2 dance to get and return an authkey. The "ephemeral"
 // property defaults to true if unspecified. The "preauthorized" defaults to
 // false. The "baseURL" defaults to
 // https://api.tailscale.com.
-func resolveAuthKey(ctx context.Context, v string) (string, error) {
+func resolveAuthKey(ctx context.Context, v, tags string) (string, error) {
```
Comment (bradfitz) [inline cmd/tailscale/cli/up.go:1130]: document this new field.

Comment (maisem) [reply cmd/tailscale/cli/up.go:1131]: Done

**Pull request merged**